airbyte-workers/src/main/java/io/airbyte/workers/protocols/Mapper.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.workers.protocols;
+
+import io.airbyte.commons.functional.CheckedConsumer;
+import io.airbyte.config.StandardSyncInput;
+import io.airbyte.config.StandardTargetConfig;
+
+public interface Mapper<T> extends CheckedConsumer<T, Exception>, AutoCloseable {
+
+  void start(StandardTargetConfig targetConfig, StandardSyncInput syncInput) throws Exception;
+
+  @Override
+  void accept(T message) throws Exception;
```

Comment:
I think this should be a function instead of a consumer. We probably don't want to do in place transformation but instead copy and return the modified copy